services/QuillLMS/app/models/user.rb

```diff
@@ -655,6 +656,18 @@ def sales_contact?
     role == SALES_CONTACT
   end
 
+  def set_time_zone
+    z = ::Ziptz.new
+    school_timezone = z.time_zone_name(school&.zipcode || school&.mail_zipcode)
+
+    if school_timezone.present?
+      self.time_zone = school_timezone
+    else
+      geocoder_results = Geocoder.search(ip_address)
```

Comment:
Are we guaranteed to have an ip_address for every user?  If not, we might want to test what happens with bad input to Geocoder.

Reply:
I'm also curious about this. What about VPNs? 

It might be better to have the user self-configure, and just message this issue louder in various places.

Reply:
@happythenewsad see response from PG [here](https://www.notion.so/Implement-automatic-timezone-setting-cd57b15b81d6430a83569fe1d5605fe7?d=eaa9aa84b46048fd994e306ac117d8e5)